In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px
%matplotlib inline

In [2]:
from plotly import __version__
from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot
print(__version__)

4.14.3


In [3]:
import plotly.graph_objects as go

In [4]:
import cufflinks as cf
init_notebook_mode(connected=True)
cf.go_offline()

In [5]:
from plotly.subplots import make_subplots

In [6]:
df = pd.read_csv('/Users/tolulopeogunfuwa/Desktop/Desktop/2021_Files/DATA ANALYTICS/PRACTISE/HAMOYE/STAGE E/bestsellers with categories.csv')

In [7]:
df.head()

Name  \
0                      10-Day Green Smoothie Cleanse   
1                                  11/22/63: A Novel   
2            12 Rules for Life: An Antidote to Chaos   
3                             1984 (Signet Classics)   
4  5,000 Awesome Facts (About Everything!) (Natio...   

                     Author  User Rating  Reviews  Price  Year        Genre  
0                  JJ Smith          4.7    17350      8  2016  Non Fiction  
1              Stephen King          4.6     2052     22  2011      Fiction  
2        Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3             George Orwell          4.7    21424      6  2017      Fiction  
4  National Geographic Kids          4.8     7665     12  2019  Non Fiction

In [8]:
df.shape

(550, 7)

In [10]:
df.Year.nunique(dropna = True)

11

**11 Years taken into account**

In [11]:
df.Author.nunique(dropna = True)

248

**248 different Authors accounted for**

In [12]:
df.Genre.nunique(dropna = True)

2

**Only Two(2) Genres listed**

In [13]:
df['Reviews'].sum()

6574305

**A total of 6,574,305 reviews across 11 years**

In [14]:
df.Name.nunique(dropna = True)

351

In [15]:
df.Name.duplicated().sum()

199

**351 Unique Books. 199 duplicates**

In [17]:
(~df.Name.duplicated()).sum()

351

In [18]:
df.loc[df.Name.duplicated(keep=False), :]

Name           Author  \
9                            A Man Called Ove: A Novel  Fredrik Backman   
10                           A Man Called Ove: A Novel  Fredrik Backman   
20                         All the Light We Cannot See    Anthony Doerr   
21                         All the Light We Cannot See    Anthony Doerr   
32                                            Becoming   Michelle Obama   
..                                                 ...              ...   
544                                             Wonder    R. J. Palacio   
546  You Are a Badass: How to Stop Doubting Your Gr...      Jen Sincero   
547  You Are a Badass: How to Stop Doubting Your Gr...      Jen Sincero   
548  You Are a Badass: How to Stop Doubting Your Gr...      Jen Sincero   
549  You Are a Badass: How to Stop Doubting Your Gr...      Jen Sincero   

     User Rating  Reviews  Price  Year        Genre  
9            4.6    23848      8  2016      Fiction  
10           4.6    23848      8  2017      Fiction  
20           4.6    36348     14  2014      Fiction  
21           4.6    36348     14  2015      Fiction  
32           4.8    61133     11  2018  Non Fiction  
..           ...      ...    ...   ...          ...  
544          4.8    21625      9  2017      Fiction  
546          4.7    14331      8  2016  Non Fiction  
547          4.7    14331      8  2017  Non Fiction  
548          4.7    14331      8  2018  Non Fiction  
549          4.7    14331      8  2019  Non Fiction  

[295 rows x 7 columns]

In [23]:
df1 = df.drop_duplicates(subset=['Name'])

In [24]:
df1.head()

Name  \
0                      10-Day Green Smoothie Cleanse   
1                                  11/22/63: A Novel   
2            12 Rules for Life: An Antidote to Chaos   
3                             1984 (Signet Classics)   
4  5,000 Awesome Facts (About Everything!) (Natio...   

                     Author  User Rating  Reviews  Price  Year        Genre  
0                  JJ Smith          4.7    17350      8  2016  Non Fiction  
1              Stephen King          4.6     2052     22  2011      Fiction  
2        Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3             George Orwell          4.7    21424      6  2017      Fiction  
4  National Geographic Kids          4.8     7665     12  2019  Non Fiction

In [25]:
df1.Name.duplicated().sum()

0

In [26]:
df1.shape

(351, 7)

In [28]:
df1['Reviews'].sum()

3439689

**After dropping the duplicates, we have a revised number of reviews. 3,439,689.**

In [33]:
df1['User Rating'].nunique(dropna = True)

14

In [34]:
df1['User Rating'].max()

4.9

In [35]:
df1['User Rating'].min()

3.3

**The maximum User Rating was 4.9, while the minimum was 3.3**

In [40]:
df1['User Rating'].value_counts()

4.8    81
4.6    69
4.7    65
4.5    39
4.9    28
4.4    26
4.3    21
4.2     8
4.1     5
4.0     4
3.9     2
3.3     1
3.8     1
3.6     1
Name: User Rating, dtype: int64

**More books have a User Rating of 4.8 than any other rating**

In [42]:
df1.head()

Name  \
0                      10-Day Green Smoothie Cleanse   
1                                  11/22/63: A Novel   
2            12 Rules for Life: An Antidote to Chaos   
3                             1984 (Signet Classics)   
4  5,000 Awesome Facts (About Everything!) (Natio...   

                     Author  User Rating  Reviews  Price  Year        Genre  
0                  JJ Smith          4.7    17350      8  2016  Non Fiction  
1              Stephen King          4.6     2052     22  2011      Fiction  
2        Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3             George Orwell          4.7    21424      6  2017      Fiction  
4  National Geographic Kids          4.8     7665     12  2019  Non Fiction

In [47]:
df1['User Rating'] = df1['User Rating'].astype(str)

In [48]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 351 entries, 0 to 546
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Name         351 non-null    object
 1   Author       351 non-null    object
 2   User Rating  351 non-null    object
 3   Reviews      351 non-null    int64 
 4   Price        351 non-null    int64 
 5   Year         351 non-null    int64 
 6   Genre        351 non-null    object
dtypes: int64(3), object(4)
memory usage: 21.9+ KB


In [52]:
df1['User Rating'].value_counts()

4.8    81
4.6    69
4.7    65
4.5    39
4.9    28
4.4    26
4.3    21
4.2     8
4.1     5
4.0     4
3.9     2
3.3     1
3.8     1
3.6     1
Name: User Rating, dtype: int64

In [53]:
df1['Genre'].value_counts()

Non Fiction    191
Fiction        160
Name: Genre, dtype: int64

In [54]:
df1['Year'].value_counts()

2009    50
2010    40
2013    33
2011    31
2018    30
2015    30
2016    28
2012    28
2019    27
2017    27
2014    27
Name: Year, dtype: int64

In [55]:
df1['Price'].sum()

4590

In [56]:
df1['Price'].max()

105

In [57]:
df1['Price'].min()

0